frame/mosaic/src/decay.rs

```diff
@@ -0,0 +1,99 @@
+use frame_support::pallet_prelude::*;
+use num_traits::{CheckedAdd, CheckedDiv, CheckedMul, CheckedSub, One, Saturating, Zero};
+
+pub trait Decayer<Balance, BlockNumber> {
+	fn checked_decay(
+		&self,
+		amount: Balance,
+		last_decay_block: BlockNumber,
+		current_block: BlockNumber,
+	) -> Option<Balance>;
+
+	fn full_recovery_period(&self, amount: Balance) -> Option<BlockNumber>;
```

Comment:
what None return means here?

Reply:
It mean that it cannot compute the recovery period, A.K.A. that the recovery < 1 block or > max_size(block)

Reply:
so it is kind of no recovery period if <1 , so it is 0 recovery period?

Reply:
i am clear about how decay works and per semantics. so it is not clear from API description.
if amount > than possible, than can mint so much so no possible recovery